src/crimson/mon/MonClient.cc

```diff
@@ -0,0 +1,477 @@
+#include "MonClient.h"
+
+#include <random>
+
+#include <seastar/core/future-util.hh>
+#include <seastar/core/lowres_clock.hh>
+#include <seastar/util/log.hh>
+
+#include "auth/AuthClientHandler.h"
+#include "auth/AuthMethodList.h"
+#include "auth/KeyRing.h"
+#include "auth/RotatingKeyRing.h"
+
+#include "crimson/auth/KeyRing.h"
+#include "crimson/common/config_proxy.h"
+#include "crimson/common/log.h"
+#include "crimson/net/Connection.h"
+#include "crimson/net/SocketMessenger.h"
+
+#include "messages/MAuth.h"
+#include "messages/MConfig.h"
+#include "messages/MLogAck.h"
+#include "messages/MMonCommand.h"
+#include "messages/MMonCommandAck.h"
+#include "messages/MMonGetVersion.h"
+#include "messages/MMonGetVersionReply.h"
+#include "messages/MMonMap.h"
+#include "messages/MMonSubscribeAck.h"
+
+namespace {
+  seastar::logger& logger() {
+    return ceph::get_logger(ceph_subsys_monc);
+  }
+}
+namespace ceph::mon {
+
+Connection::Connection(ceph::net::ConnectionRef conn,
+                       KeyRing* keyring)
+  : conn{conn},
+    rotating_keyring{nullptr, CEPH_ENTITY_TYPE_OSD, keyring}
+{}
+
+seastar::future<> Connection::handle_auth_reply(Ref<MAuthReply> m)
+{
+  reply.set_value(m);
+  return seastar::make_ready_future<>();
+}
+
+seastar::future<> Connection::renew_tickets()
+{
+  if (auth->need_tickets()) {
+    return send_auth_request();
+  } else {
+    return seastar::make_ready_future<>();
+  }
+}
+
+std::unique_ptr<AuthClientHandler>
+Connection::create_auth(Ref<MAuthReply> m,
+                        const EntityName& name,
+                        uint32_t want_keys)
+{
+  std::unique_ptr<AuthClientHandler> auth;
+  auth.reset(AuthClientHandler::create(nullptr,
```

Comment:
`AuthClientHandler` queries config, for instance, for the name of the keyring, etc. so we need a way to fix the calls like `cct->_conf->name` in `CephxClientHandler.cc` .